src/components/PersonItem/PersonItem.tsx

```diff
@@ -0,0 +1,43 @@
+import { useParams } from 'react-router-dom';
+import { Person } from '../../types';
+import { PersonLink } from '../PersonLink';
+import classNames from 'classnames';
+import { useContext } from 'react';
+import { PeopleContext } from '../../peopleContext';
+import { getPersonByName } from '../../utils/getPeopleByName';
+
+type Props = {
+  person: Person;
+};
+
+export const PersonItem: React.FC<Props> = ({ person }) => {
+  const { slug: userId, sex, born, died, fatherName, motherName } = person;
+  const { slug: selectedUserId } = useParams();
+
+  const people = useContext(PeopleContext);
+  const mother = getPersonByName(motherName, people);
```

Comment:
In the current implementation, you're calling `getPersonByName` function for each rendered `PersonItem`. This could potentially lead to performance issues when dealing with a large number of people. Instead of fetching the mother and father for each person individually, consider fetching all people once and storing them in a data structure that allows for quick access by name. This way, you can avoid unnecessary function calls and improve the performance of your application.